### PR TITLE
Document declare cheatcode in cairo1

### LINKED ISSUE #1666: Document `declare` cheatcode in cairo1
Add documentation similar to https://docs.swmansion.com/protostar/docs/tutorials/testing/cheatcodes

### PR DESCRIPTION
**Pull request by cptartur**
Closes #1666

### CODE REVIEW

File: website/docs/tutorials/07-testing/02-cheatcodes/declare-cairo0.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+```python
+def declare(contract: str) -> DeclaredContract:
+
+class DeclaredContract:
+    class_hash: int
+```
+Declares contract given a path relative to a Protostar project root.
+
+- `contract` — A path to the contract's main `.cairo` source file.
```
Comment (Arcticae) [inline website/docs/tutorials/07-testing/02-cheatcodes/declare-cairo0.md:11]: Shouldn't this be contract's name from `protostar.toml` only?

File: website/docs/tutorials/07-testing/02-cheatcodes/declare-cairo0.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# `declare_cairo0`
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/07-testing/02-cheatcodes/declare-cairo0.md:1]: I don't think it should be included here, we should create corresponding section in cairo1 support section. Cheatcode should be documented in rust. 

Contract based test runner has its own declare and does not support anything cairo1 related.

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Cheatcodes
+
+Most of the time, testing smart contracts with assertions only is not enough. Some test cases require manipulating the
+state of the blockchain, as well as checking for reverts and events. For that reason, Protostar provides a set of
+cheatcodes.
+
+Cheatcodes are available as normal cairo functions. You don't have to import anything. When Protostar runs tests,
+it takes care of replacing this functions with code necessary to modify the state of the network.
```
Comment (Arcticae) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/README.md:8]:
```suggestion
Cheatcodes are available as normal cairo functions. You don't have to import anything.

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare-cairo0.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# `declare_cairo0`
+
+```cairo
+fn declare_cairo0(contract: felt) -> Result::<felt, felt> nopanic;
+```
+Declares a Cairo0 (old syntax) contract given a contract name defined the [protostar.toml](../../04-configuration-file.md) configuration
+file.
```
Comment (Arcticae) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare-cairo0.md:7]:
```suggestion
Declares a Cairo0 (old syntax) contract given a contract name defined the [protostar.toml](../../04-configuration-file.md) configuration file.
```

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# `declare`
+
+```cairo
+fn declare(contract: felt) -> Result::<felt, felt> nopanic;
+```
+
+Declares a contract given a contract name defined the [protostar.toml](../../04-configuration-file.md) configuration
```
Comment (Arcticae) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md:7]: We should tell the user that this cheatcode only supports cairo 1 now, and reference to declaration of old ones.

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+file.
+
+- `contract` name of the contract as cairo shortstring (e.g. `declare('mycontract')`).
```
Comment (Arcticae) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md:10]: We should add an example for each of those, we had those in previous docs

**Review by Arcticae — CHANGES_REQUESTED** [review]

**Comment by MaksymilianDemitraszek** [comment]
Can we sync it with master?

**Comment by cptartur** [comment]
> Can we sync it with master?

After #1625 is merged I'll sync it.

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/README.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
+# Cheatcodes
+
+Most of the time, testing smart contracts with assertions only is not enough. Some test cases require manipulating the
+state of the blockchain, as well as checking for reverts and events. For that reason, Protostar provides a set of
+cheatcodes.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/README.md:5]: This does not have much sense in current setting. You are basically unable to test anything contract related without cheatcodes

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+# `declare`
+
+```cairo
+fn declare(contract: felt) -> Result::<felt, felt> nopanic;
+```
+
+Declares a contract given a contract name defined in the [protostar.toml](../../04-configuration-file.md) configuration
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md:7]:
```suggestion
Declares a contract given its name defined in the [protostar.toml](../../04-configuration-file.md) configuration
```

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+file.
+
+- `contract` name of the contract as cairo shortstring (e.g. `declare('mycontract')`).
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare.md:10]:
```suggestion
- `contract` name of a contract as cairo shortstring (e.g. `declare('mycontract')`).
```

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare-cairo0.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# `declare_cairo0`
+
+```cairo
+fn declare_cairo0(contract: felt) -> Result::<felt, felt> nopanic;
+```
+Declares a Cairo0 (old syntax) contract given a contract name defined in the [protostar.toml](../../04-configuration-file.md) configuration file.
+
+- `contract` name of the contract as cairo shortstring (e.g. `declare_cairo0('mycontract')`).
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/08-cairo-1-support/03-cheatcodes/declare-cairo0.md:8]:
```suggestion
- `contract` name of a contract as cairo shortstring (e.g. `declare_cairo0('mycontract')`).
```

File: website/docs/tutorials/08-cairo-1-support/03-cheatcodes/README.md (hunk 7 above)
Comment (cptartur) [reply website/docs/tutorials/08-cairo-1-support/03-cheatcodes/README.md:5]: What do you mean exactly? I basically copied the description from the current documentation. I can rewrite it but not sure what should be changed

**Review by Arcticae — APPROVED** [review]

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Pull request merged**